### PR TITLE
JAMES-2648 cassandra list sources support implementation

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTable.java [hunk 1]
```diff
@@ -115,15 +113,15 @@ public TooManyMappingException(String string) {
...
+    default Stream<MappingSource> listSources(Mapping mapping) throws RecipientRewriteTableException {
         Preconditions.checkArgument(listSourcesSupportedType.contains(mapping.getType()),
             String.format("Not supported mapping of type %s", mapping.getType()));
 
         return getAllMappings()
             .entrySet().stream()
             .filter(entry -> entry.getValue().contains(mapping))
             .map(Map.Entry::getKey)
-            .collect(Guavate.toImmutableList());
+            .sorted(Comparator.comparing(MappingSource::asMailAddressString));
```
Comment (chibenwa) [inline server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTable.java]: If we sort it, then we have to process all sources, which defeats the performance optimisation enabled by the stream optimisation.

Maybe we need to sort 'only before collecting' and delegate that responsibility to the caller?

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java [hunk 2]
```diff
@@ -80,4 +92,20 @@ protected Mappings mapAddress(String user, Domain domain) {
             () -> cassandraRecipientRewriteTableDAO.retrieveMappings(MappingSource.fromDomain(domain)).blockOptional())
                 .orElse(MappingsImpl.empty());
     }
+
+    @Override
+    public Stream<MappingSource> listSources(Mapping mapping) throws RecipientRewriteTableException {
+        Preconditions.checkArgument(listSourcesSupportedType.contains(mapping.getType()),
+            String.format("Not supported mapping of type %s", mapping.getType()));
+
+        final SchemaVersion schemaVersion = cassandraSchemaVersionDAO.getCurrentSchemaVersion()
```
Comment (chibenwa) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java]: We don't use `final` for variables

File: server/data/data-cassandra/src/test/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableTest.java [hunk 3]
```diff
@@ -61,11 +79,60 @@ public static void tearDownClass() {
...
+        CassandraRecipientRewriteTable rrt = new CassandraRecipientRewriteTable(
+            cassandraRecipientRewriteTableDAO,
+            cassandraMappingsSourcesDAO,
+            cassandraSchemaVersionDAO);
         rrt.configure(new DefaultConfigurationBuilder());
         return rrt;
     }
 
+    @Test
+    public void listSourcesShouldUseDefaultMethodWhenMappingsSourcesIsUnsupported() throws Exception {
```
Comment (chibenwa) [inline server/data/data-cassandra/src/test/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableTest.java]: We don't want specific tests IMO.

Maybe we do want to run `all tests` for version 6 and version 7

File: server/data/data-cassandra/src/test/java/org/apache/james/rrt/cassandra/CassandraStepdefs.java [hunk 4]
```diff
@@ -39,7 +42,9 @@ public CassandraStepdefs(RewriteTablesStepdefs mainStepdefs) {
 
     @Before
     public void setup() throws Throwable {
-        cassandra = CassandraCluster.create(CassandraRRTModule.MODULE, RewriteTablesTest.cassandraServer.getHost());
+        cassandra = CassandraCluster.create(
+            CassandraModule.aggregateModules(CassandraRRTModule.MODULE, CassandraSchemaVersionModule.MODULE),
```
Comment (blackheaven) [inline server/data/data-cassandra/src/test/java/org/apache/james/rrt/cassandra/CassandraStepdefs.java]: Can you extract it in a `MODULE` constant please?

**Comment by Arsnael** [comment]
I think not the first time I see something like this, not sure what to think of it : 
```
[8641ad617944cb616db298be5134c4a288f24c56] [ERROR]   WebAdminServerIntegrationTest.postShouldDoMigrationAndUpdateToTheLatestVersion:282 1 expectation failed.
[8641ad617944cb616db298be5134c4a288f24c56] Response body doesn't match expectation.
[8641ad617944cb616db298be5134c4a288f24c56] Expected: is "{\"version\":7}"
[8641ad617944cb616db298be5134c4a288f24c56]   Actual: {"version":5}
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**